### ISSUE TITLE
How should I customize lldb?

### ISSUE THREAD

**Issue by elipeng** [PRESIDENT810/slowDebugTest#3]
Currently, the problem of slow debugging has been reproduced through this project.

The modification you suggested here is to skip the unnecessary dlopen step. I understand that you should compile an lldb yourself and then replace it in this project, right?

It's unclear how to replace it.

**Comment by kainzhong** [comment]
I'm using MacOS so I fetched Apple's LLDB fork, compiled a LLDB.framework and replaced `/Applications/Xcode.app/Contents/SharedFrameworks/LLDB.framework` with it.

Btw I'm not sure if the unnecessary dlopen step has been fixed by Apple since I haven't touched this for years.

**Comment by elipeng** [comment]
> I'm using MacOS so I fetched Apple's LLDB fork, compiled a LLDB.framework and replaced `/Applications/Xcode.app/Contents/SharedFrameworks/LLDB.framework` with it.
> 
> Btw I'm not sure if the unnecessary dlopen step has been fixed by Apple since I haven't touched this for years.

Yes, I also use macOS, and I have compiled the lldb executable file through the LLVM source code;

If I directly replace LLDB in Xcode.app, this will affect all projects. Currently, I am trying to use the ~/.lldbinit file to specify lldb, but currently I cannot write the correct statement.

In addition, I would like to ask about the remarks about sleep in the source code: // wait so I have time to let debugger attach to process lldb-rpc-server

Is there anything I need to pay attention to here?

**Comment by kainzhong** [comment]
> If I directly replace LLDB in Xcode.app, this will affect all projects. Currently, I am trying to use the ~/.lldbinit file to specify lldb, but currently I cannot write the correct statement.

I just replaced LLDB.framework and affect all projects. I'm not aware if there's a way to select which LLDB.framework to use via lldbinit file (maybe there is but I don't really know)

> In addition, I would like to ask about the remarks about sleep in the source code: // wait so I have time to let debugger attach to process lldb-rpc-server

I was using debugger to debug lldb-rpc-server (which loads the LLDB dynamic framework) using attach mode, so I need to make it sleep a while so I have time to launch the debugger and attach to the lldb-rpc-server process.

**Comment by elipeng** [comment]
> > 如果我直接替换Xcode.app中的LLDB，这将影响所有项目。目前，我正在尝试使用 ~/.lldbinit 文件来指定 lldb，但目前我无法编写正确的语句。
> 
> 我刚刚替换了 LLDB.framework 并影响了所有项目。我不知道是否有办法通过 lldbinit 文件选择要使用的 LLDB.framework （也许有，但我真的不知道）
> 
> > 另外，我想问一下源码中关于sleep的备注： // wait so I have time to let debugger Attach to process lldb-rpc-server
> 
> 我正在使用调试器使用附加模式调试 lldb-rpc-server （加载 LLDB 动态框架），因此我需要让它休眠一段时间，以便我有时间启动调试器并附加到 lldb-rpc-server 进程。

Thank you very much. I will try to replace the global LLDB.framework first.

**Comment by elipeng** [comment]
> > If I directly replace LLDB in Xcode.app, this will affect all projects. Currently, I am trying to use the ~/.lldbinit file to specify lldb, but currently I cannot write the correct statement.
> 
> I just replaced LLDB.framework and affect all projects. I'm not aware if there's a way to select which LLDB.framework to use via lldbinit file (maybe there is but I don't really know)
> 
> > In addition, I would like to ask about the remarks about sleep in the source code: // wait so I have time to let debugger attach to process lldb-rpc-server
> 
> I was using debugger to debug lldb-rpc-server (which loads the LLDB dynamic framework) using attach mode, so I need to make it sleep a while so I have time to launch the debugger and attach to the lldb-rpc-server process.

Thank you very much. I will try to replace the global LLDB.framework first.
